test_integration/test_plugin_static_policy.py

```diff
@@ -93,24 +127,31 @@ def setUpClass(cls):
         cls._geopmadminagent = getSystemConfigAgent()
         cls._geopmadminagentpolicy = getSystemConfigPolicy()
 
-    @unittest.skipUnless(getSystemConfigAgent() in ['frequency_map', 'energy_efficient'],
-                         'Requires environment default/override to be configured to cap frequency.')
+    @skip_unless_energy_agent()
     def test_frequency_cap_enforced(self):
+        policy_name = None
         if self._geopmadminagent == 'frequency_map':
-            test_freq = self._geopmadminagentpolicy['FREQ_MAX']
+            policy_name = 'FREQ_MAX'
         elif self._geopmadminagent == 'energy_efficient':
-            test_freq = self._geopmadminagentpolicy['FREQ_FIXED']
-        current_freq = geopmpy.pio.read_signal("MSR::PERF_CTL:FREQ", "board", 0)
-        self.assertEqual(test_freq, current_freq)
+            policy_name = 'FREQ_FIXED'
+        try:
+            test_freq = self._geopmadminagentpolicy[policy_name]
+            current_freq = geopmpy.pio.read_signal("MSR::PERF_CTL:FREQ", "board", 0)
+            self.assertEqual(test_freq, current_freq)
+        except KeyError:
+            self.skipTest('Expected frequency cap "{}" for agent missing from policy'.format(policy_name))
```

Comment:
geopmadmin will call validate policy given policy that is configured.  these key error checks are redundant and I think should be removed, no?

Reply:
Is the plan that we will have some required policy fields?  Currently having a min frequency only is considered a valid policy from the agent's point of view.